gluten-substrait/src/main/scala/org/apache/spark/sql/execution/GlutenAutoAdjustStageResourceProfile.scala

```diff
@@ -156,17 +158,25 @@ object GlutenAutoAdjustStageResourceProfile extends Logging {
   def applyNewResourceProfileIfPossible(
       plan: SparkPlan,
       rp: ResourceProfile,
-      rpManager: ResourceProfileManager): SparkPlan = {
+      rpManager: ResourceProfileManager,
+      sparkConf: SparkConf): SparkPlan = {
+    // Reflects resource changes in some configurations that will be passed to the native side.
+    val coresPerExecutor = rp.getExecutorCores.getOrElse(sparkConf.get(EXECUTOR_CORES))
+    val coresPerTask = rp.getTaskCpus.getOrElse(sparkConf.get(CPUS_PER_TASK))
+    val taskSlots = coresPerExecutor / coresPerTask
+    SQLConf.get.setConfString(GlutenCoreConfig.NUM_TASK_SLOTS_PER_EXECUTOR.key, taskSlots.toString)
+    val offHeapSize = rp.executorResources.get(ResourceProfile.OFFHEAP_MEM).map(_.amount).get
+    SQLConf.get.setConfString(
+      GlutenCoreConfig.COLUMNAR_OFFHEAP_SIZE_IN_BYTES.key,
+      offHeapSize.toString)
+    SQLConf.get.setConfString(
+      GlutenCoreConfig.COLUMNAR_TASK_OFFHEAP_SIZE_IN_BYTES.key,
+      (offHeapSize / taskSlots).toString)
```

Comment:
This config should be reset to original value if stage still use the default resource profile after some stage hit the rule here.